### PR TITLE
Update the Trace function to return a Void type

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
The Trace function doesn't return anything. For legacy reasons, it currently returns DType.Boolean (when it was created, we didn't have DType.Void). This PR updates it to return Void instead.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests.Shared/TexlTests.cs [hunk 1]
```diff
@@ -2693,6 +2693,25 @@ public void TexlFunctionTypeSemanticsTable_Negative(string script, string expect
...
+        [InlineData("Trace(\"hello\")")]
+        [InlineData("Trace(\"hello\", TraceSeverity.Warning)")]
+        [InlineData("Trace(\"hello\", TraceSeverity.Warning, { a: 1 })")]
+        public void TexlFunctionTypeSemanticsTrace(string expression)
+        {
+            foreach (var powerFxV1 in new[] { false, true })
+            {
+                var features = powerFxV1 ? Features.PowerFxV1 : Features.None;
+                var engine = new Engine(new PowerFxConfig(features));
+                var options = new ParserOptions() { NumberIsFloat = true, AllowsSideEffects = true };
```
Comment (anderson-joyle) [inline src/tests/Microsoft.PowerFx.Core.Tests.Shared/TexlTests.cs:2706]: >NumberIsFloat = true [](http://example.com/codeflow?start=52&length=20)

Do we need this parse option to run the test?

**Review by anderson-joyle — DISMISSED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Core.Tests.Shared/TexlTests.cs (hunk 1 above)
Comment (CarlosFigueiraMSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests.Shared/TexlTests.cs]: Good point, it's not necessary. Removed.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by anderson-joyle — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**